### ISSUE TITLE
 NoMethodError in StatusesController#create 

### ISSUE THREAD

**Issue by shawnpyle** [bterlson/redmine-status-updates#3]
I get the following error when trying to update my status, either within a project or from the Update status link at the very top.

<pre>
 NoMethodError in StatusesController#create 
undefined method `members' for #<Project:0x8b35170>
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/attribute_methods.rb:256:in `method_missing'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/associations/association_proxy.rb:173:in `send'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/associations/association_proxy.rb:173:in `method_missing'
vendor/plugins/redmine_status/app/models/status.rb:63:in `recipients'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/core_ext/object/misc.rb:28:in `returning'
vendor/plugins/redmine_status/app/models/status.rb:62:in `recipients'
vendor/plugins/redmine_status/app/models/status_mailer.rb:3:in `realtime_notification'
/opt/local/lib/ruby/gems/1.8/gems/actionmailer-2.1.2/lib/action_mailer/base.rb:446:in `__send__'
/opt/local/lib/ruby/gems/1.8/gems/actionmailer-2.1.2/lib/action_mailer/base.rb:446:in `create!'
/opt/local/lib/ruby/gems/1.8/gems/actionmailer-2.1.2/lib/action_mailer/base.rb:439:in `initialize'
/opt/local/lib/ruby/gems/1.8/gems/actionmailer-2.1.2/lib/action_mailer/base.rb:383:in `new'
/opt/local/lib/ruby/gems/1.8/gems/actionmailer-2.1.2/lib/action_mailer/base.rb:383:in `method_missing'
vendor/plugins/redmine_status/app/models/status_observer.rb:5:in `after_create'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/observer.rb:171:in `send'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/observer.rb:171:in `update'
/opt/local/lib/ruby/1.8/observer.rb:185:in `notify_observers'
/opt/local/lib/ruby/1.8/observer.rb:184:in `each'
/opt/local/lib/ruby/1.8/observer.rb:184:in `notify_observers'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/callbacks.rb:309:in `notify'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/callbacks.rb:296:in `callback'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/callbacks.rb:221:in `create_without_timestamps'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/timestamp.rb:29:in `create'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/base.rb:2483:in `create_or_update_without_callbacks'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/callbacks.rb:207:in `create_or_update'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/base.rb:2211:in `save_without_validation'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/validations.rb:911:in `save_without_dirty'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/dirty.rb:75:in `save_without_transactions'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:106:in `save'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/connection_adapters/abstract/database_statements.rb:66:in `transaction'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:79:in `transaction'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:98:in `transaction'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:106:in `save'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:118:in `rollback_active_record_state!'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/transactions.rb:106:in `save'
vendor/plugins/redmine_status/app/controllers/statuses_controller.rb:28:in `create'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/base.rb:1166:in `send'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/base.rb:1166:in `perform_action_without_filters'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/filters.rb:579:in `call_filters'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/filters.rb:572:in `perform_action_without_benchmark'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/benchmarking.rb:68:in `perform_action_without_rescue'
/opt/local/lib/ruby/1.8/benchmark.rb:293:in `measure'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/benchmarking.rb:68:in `perform_action_without_rescue'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/rescue.rb:201:in `perform_action_without_caching'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/caching/sql_cache.rb:13:in `perform_action'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/connection_adapters/abstract/query_cache.rb:33:in `cache'
/opt/local/lib/ruby/gems/1.8/gems/activerecord-2.1.2/lib/active_record/query_cache.rb:8:in `cache'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/caching/sql_cache.rb:12:in `perform_action'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/base.rb:529:in `send'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/base.rb:529:in `process_without_filters'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/filters.rb:568:in `process_without_session_management_support'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/session_management.rb:130:in `process'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/base.rb:389:in `process'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:149:in `handle_request'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:107:in `dispatch'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:104:in `synchronize'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:104:in `dispatch'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:120:in `dispatch_cgi'
/opt/local/lib/ruby/gems/1.8/gems/actionpack-2.1.2/lib/action_controller/dispatcher.rb:35:in `dispatch'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/rails.rb:76:in `process'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/rails.rb:74:in `synchronize'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/rails.rb:74:in `process'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:159:in `process_client'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:158:in `each'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:158:in `process_client'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:285:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:285:in `initialize'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:285:in `new'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:285:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:268:in `initialize'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:268:in `new'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel.rb:268:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/configurator.rb:282:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/configurator.rb:281:in `each'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/configurator.rb:281:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/bin/mongrel_rails:128:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/lib/mongrel/command.rb:212:in `run'
/opt/local/lib/ruby/gems/1.8/gems/mongrel-1.1.5/bin/mongrel_rails:281
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:503:in `load'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:503:in `load'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:355:in `new_constants_in'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:503:in `load'
/opt/local/lib/ruby/gems/1.8/gems/rails-2.1.2/lib/commands/servers/mongrel.rb:64
/opt/local/lib/ruby/vendor_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
/opt/local/lib/ruby/vendor_ruby/1.8/rubygems/custom_require.rb:31:in `require'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:510:in `require'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:355:in `new_constants_in'
/opt/local/lib/ruby/gems/1.8/gems/activesupport-2.1.2/lib/active_support/dependencies.rb:510:in `require'
/opt/local/lib/ruby/gems/1.8/gems/rails-2.1.2/lib/commands/server.rb:39
/opt/local/lib/ruby/vendor_ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require'
/opt/local/lib/ruby/vendor_ruby/1.8/rubygems/custom_require.rb:31:in `require'
script/server:3

Request

Parameters:

{"commit"=>"Save",
 "authenticity_token"=>"4f0cda9b2360da4e87fb0147bd163cd25c4b98d8",
 "status"=>{"project_id"=>"8",
 "message"=>"this is a test"}}

Show session dump
--- 
adminprojects_sort: 
  :key: name
  :order: asc
:user_id: 7
flash: !map:ActionController::Flash::FlashHash {}
usersindex_sort: 
  :key: login
  :order: asc

Headers:
{"cookie"=>[],  "Cache-Control"=>"no-cache"}
</pre>

It appears to not be saving the status record properly and therefore isn't able to get the project (and it's members).

**Comment by purushottamroy** [comment]
even i m getting the same error as above , any help will be appreciated, thanks

undefined method `members' for #ActiveRecord::Associations::BelongsToAssociation:0xb64a4028

RAILS_ROOT: /home/icicle/roysites/redmine
Application Trace | Framework Trace | Full Trace

Request

Parameters:

{"commit"=>"Save",
 "authenticity_token"=>"Rq2MhZKKQ092epihnGqnQ6by72MpQS4fgNV+7bhNobo=",
 "status"=>{"project_id"=>"14",
 "message"=>"testing status update plugin"}}

Show session dump
Response

Headers:

{"Content-Type"=>"",
 "Cache-Control"=>"no-cache"}

**Comment by JangoSteve** [comment]
I had this problem too. Seems to have been caused by `attr_protected :project_id`, though I'm not sure why, because the `status.project_id` was being set explicitly. I'm going to chalk it up to an old Rails 2.3 bug perhaps.

I was able to fix it by removing `attr_projected :project_id`. This isn't generally advised and leaves a security vulnerability in terms of allowing people to post statuses to projects they don't belong to.

Just to be safe, I added `Status.new(params[:status].reject{ |k,v| k == :project_id })` to manually remove `:project_id` from the mass assignment params, even though it wasn't needed since `status.project_id` is being overridden by the `check_project_id` method.

Once I fixed that, I got another exception on the /statuses page due to incorrect project identifier usage, which I also ended up fixing.

TL;DR I fixed these two issues in the [create-status-fix branch of my fork](https://github.com/JangoSteve/redmine-status-updates/tree/create-status-fix).
